entities/src/main/java/com/flipkart/varadhi/entities/TopicResource.java

```diff
@@ -29,4 +45,13 @@ public TopicResource(
         this.capacityPolicy = capacityPolicy;
     }
 
+    public void validate() {
+        Set<ConstraintViolation<TopicResource>> violations = super.getValidator().validate(this);
+        if (violations.isEmpty()) {
+            return;
+        }
+        throw new ValidationException(violations.stream()
```

Comment:
IllegalArgumentException might be better here.

Reply:
moved